### PR TITLE
fix(gas_price_service_v1): ensure that the algo updater is synced to l2 block height

### PR DESCRIPTION
**Pull request by rymnc**
## Linked Issues/PRs

- none

## Description

We didn't have this check in the `uninitialized_task__init__if_metadata_behind_l2_height_then_sync` test, which caused us to miss the duped algo updater.

## Checklist
- [ ] Breaking changes are clearly marked as such in the PR description and changelog
- [ ] New behavior is reflected in tests
- [ ] [The specification](https://github.com/FuelLabs/fuel-specs/) matches the implemented behavior (link update PR if changes are needed)

### Before requesting review
- [x] I have reviewed the code myself
- [ ] I have created follow-up issues caused by this PR and linked them here

### After merging, notify other teams

[Add or remove entries as needed]

- [ ] [Rust SDK](https://github.com/FuelLabs/fuels-rs/)
- [ ] [Sway compiler](https://github.com/FuelLabs/sway/)
- [ ] [Platform documentation](https://github.com/FuelLabs/devrel-requests/issues/new?assignees=&labels=new+request&projects=&template=NEW-REQUEST.yml&title=%5BRequest%5D%3A+) (for out-of-organization contributors, the person merging the PR will do this)
- [ ] Someone else?

### CODE REVIEW

File: crates/services/gas_price_service/src/v1/tests.rs [hunk 1]
```diff
@@ -779,13 +776,19 @@ async fn uninitialized_task__init__if_metadata_behind_l2_height_then_sync() {
...
     .unwrap();
 
     // when
-    service.init(&StateWatcher::started()).await.unwrap();
+    let gas_price_service = service.init(&StateWatcher::started()).await.unwrap();
 
     // then
-    // no panic
+    // sleep to allow the service to sync
+    tokio::time::sleep(std::time::Duration::from_millis(100)).await;
```
Comment (acerone85) [inline crates/services/gas_price_service/src/v1/tests.rs:788]: personally I'm not a fan of sleep of sleeps to wait for a condition to happen, but this is orthogonal to the PR.

**Review by acerone85 — APPROVED** [review]

File: crates/services/gas_price_service/src/v1/tests.rs (hunk 1 above)
Comment (netrome) [reply crates/services/gas_price_service/src/v1/tests.rs:788]: Agree, but with tokio time I can accept it since tokio auto-advances time when all futures are pending.

(Reference https://docs.rs/tokio/latest/tokio/time/fn.pause.html)

**Review by netrome — APPROVED** [review]
